src/main/java/lotto/LottoNumber.java

```diff
@@ -0,0 +1,73 @@
+package lotto;
+
+import java.util.Objects;
+
+public class LottoNumber implements Comparable<LottoNumber> {
+
+  public static final int MIN_NUMBER = 1;
+  public static final int MAX_NUMBER = 45;
+
+  private static final String ERROR_OUT_OF_RANGE = "로또 번호는 1~45 범위의 숫자여야 한다";
+
+  private static final LottoNumber[] CACHE = new LottoNumber[MAX_NUMBER + 1];
```

Comment:
```suggestion
  private static final Map<Integer, LottoNumber> CACHE = new HashMap<>();
```
배열보다 Map으로 구현하는 것은 어떨까?
정말 특별한 경우가 아니면 가능하면 배열 사용하는 것을 추천하지 않음
보통 캐싱의 경우 Map을 사용함